sample_information_validator.py

```diff
@@ -23,78 +23,280 @@
                     fmt='%(asctime)s %(levelname)s %(message)s')
 
 
-class NumericValidator(object):
-    """NumericValidator
+class ProjectSheet:
+#Class Attributes
+    SHEET_NAME = 'Sample information'
+    FIRST_LINE = 20  # First line where user submitted data is located
+    SAMPLE_NAME_COL = 'O'
+    A_RATIO_COL = 'R'  # A260:A280 ratio
+    CONC_COL = 'S'
+    VOL_COL = 'T'
+    RIN_COL = 'V'
+    SAMPLE_TYPE = 'O8'
+    PLATE_ID = 'M6'
+    PROJECT_NAME_USER_SHEET = 'M3'
 
-    Checks whether cell is numeric or not.
-    Takes optional params:
-       decimal_comma - Checks whether swedish comma is used or not.
-                       [default False]
-    """
+    # Initializer / Instance attributes
+    def __init__(self, sample_info_sheet):
+        self.sample_info_sheet = sample_info_sheet
+        self.work_sheet = None
+        self.sample_rec = None
 
-    def validate(self, cell):
-        """Checks whether value is numeric or not."""
-        logging.debug(cell.value, cell.data_type)
-        if cell.data_type != 'n':
+    # instance methods
+    def getAccessUserSheet(self):
+        if self.work_sheet == None:
+            wb = load_workbook(self.sample_info_sheet, read_only=True, data_only=True)
+            ws = wb[ProjectSheet.SHEET_NAME]
+            self.work_sheet = ws
+
+    def projectID(self):
+        self.getAccessUserSheet()
+        plate_id = self.work_sheet[ProjectSheet.PLATE_ID].value
+        project_id_user = re.findall('P[0-9]+', plate_id)[0]
+        return(project_id_user)
+
+    def getSamples(self):
+        """ identifies the all rows containing a sample name, discards rows without entry.
+        Rows containing whitespace only trigger a warning and are discarded for subsequent
+        tests """
+        cellID_withSample = list()
+        cellID_noSample = list()
+        for i in range(ProjectSheet.FIRST_LINE, ProjectSheet.FIRST_LINE+96):
```

Comment:
The `+96` looks like hard coded thing. Can there not be more than 96 rows? Or at least keep it as a class variable so that it's easier to find when one needs to change it.

Reply:
There should never be more than 96 since we work in 96 well plates and each plate has it's own sample sheet.